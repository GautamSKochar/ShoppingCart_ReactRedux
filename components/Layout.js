import React from 'react';
import Header from './Header';
import { BrowserRouter as Router } from 'react-router-dom';
import Routes from '../routes/Routes';

export default class Layout extends React.Component {
    render() {
        return (
            <div>
                <Header />
                <Routes />
                <footer>
                    <div style={{ backgroundColor: 'black' }}></div>
                </footer>
            </div>
        );
    }
}