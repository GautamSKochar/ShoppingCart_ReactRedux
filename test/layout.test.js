import React from "react";
import Layout from "../components/Layout";
import { shallow , mount, render} from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json';
import configureMockStore from "redux-mock-store";

const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}
configure({ adapter: new Adapter() });
const store = configureMockStore({});

describe('Layout Component', () => {
    it('Snapshot Test', () => {
        const component = shallow(<Layout />);
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    })
});