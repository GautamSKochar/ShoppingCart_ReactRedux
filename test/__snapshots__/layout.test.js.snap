// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Snapshot Test 1`] = `
<div>
  <Connect(Header) />
  <Routes />
  <footer>
    <div
      style={
        Object {
          "backgroundColor": "black",
        }
      }
    />
  </footer>
</div>
`;
